src/plugins/auto/src/auto_executable_network.cpp

```diff
@@ -145,9 +145,20 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
             return decltype(ov::optimal_number_of_infer_requests)::value_type {res};
         }
         if (_autoSchedule->_loadContext[ACTUALDEVICE].isAlready) {
-            real = _autoSchedule->_loadContext[ACTUALDEVICE].
-                executableNetwork->GetMetric(name).as<unsigned int>();
+            real = _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name).as<unsigned int>();
         } else {
+            auto supportedProperties =
+                _autoSContext->_core->get_property(_autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName,
+                                                   ov::supported_properties);
+            auto nireqIter =
+                std::find(supportedProperties.begin(), supportedProperties.end(), ov::optimal_number_of_infer_requests);
+            if (nireqIter != supportedProperties.end()) {
+                real =
+                    _autoSContext->_core->get_property(_autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName,
```

Comment:
Need to pass the properties which are to be passed to compile_model

Reply:
Please add test cases.

Reply:
Updated.